RecoTracker/LSTCore/src/alpaka/Event.dev.cc

```diff
@@ -194,11 +193,15 @@ void Event::addPixelSegmentToEvent(std::vector<unsigned int> const& hitIndices0,
     *nTotalMDs_buf_h.data() += n_max_pixel_md_per_modules;
     unsigned int nTotalMDs = *nTotalMDs_buf_h.data();
 
-    mdsInGPU_.emplace();
-    miniDoubletsBuffers_.emplace(nTotalMDs, nLowerModules_, devAcc_, queue_);
-    mdsInGPU_->setData(*miniDoubletsBuffers_);
+    std::array<int, 2> const mds_sizes{{static_cast<int>(nTotalMDs), static_cast<int>(nLowerModules_ + 1)}};
```

Comment:
the casting to int looked odd at first, but now I see that upstream https://github.com/cms-sw/cmssw/blob/master/DataFormats/Portable/interface/PortableDeviceCollection.h#L105
has `  using SizesArray = std::array<int32_t, members_>;` 
hopefully this gets fixed in CMSSW at some point

Reply:
> hopefully this gets fixed in CMSSW at some point

You could open an issue an explain why you think the current code is not correct.

Reply:
it sounds like negative sizes are deliberately allowed and I'd need to argue around that